api/src/main/java/io/grpc/ClientStreamTracer.java

```diff
@@ -27,6 +27,11 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2861")
 @ThreadSafe
 public abstract class ClientStreamTracer extends StreamTracer {
+  /**
+   * The call was delayed due to waiting for name resolution result.
+   */
+  public static final CallOptions.Key<Boolean> isResolutionDelay =
```

Comment:
ALL_UPPER_CASE

What object/thing "is" the resolution delay? I could understand RESOLUTION_DELAYED, or WAS_DELAYED_BY_RESOLUTION, and the like. And we should probably include "name" in resolution, to be clear.